engine/common/synchronization/engine.go

```diff
@@ -358,16 +358,22 @@ func (e *Engine) pollHeight() {
 	head := e.finalizedHeader.Get()
 	participants := e.participantsProvider.Identifiers()
 
+	nonce, err := rand.Uint64()
+	if err != nil {
+		e.log.Warn().Err(err).Msg("nonce generation failed during pollHeight")
+		return
+	}
```

Comment:
Likewise for this file as well, please add:
- error handling for the random methods, as necessary
- TODO comments for logged and swallowed errors

Reply:
https://github.com/onflow/flow-go/pull/4258/commits/7c1d3d9c89b0dff40e17882828d19f0d85d62333